File: ex-14/ex-14_java/CircleShape.java
public class CircleShape {
    private float raio;
    private Cor corPreenchimento;
    private Vetor2f origem;
    private Vetor2f posicao;

    public CircleShape(float raio) {
        this.raio = raio;
        this.corPreenchimento = new Cor(255, 255, 255);
        this.origem = new Vetor2f(0, 0);
        this.posicao = new Vetor2f(0, 0);
    }

    public void setFillColor(Cor cor) {
        this.corPreenchimento = cor;
    }

    public void setOrigin(Vetor2f origem) {
        this.origem = origem;
    }

    public void setPosition(Vetor2f posicao) {
        this.posicao = posicao;
    }

    public float getRaio() {
        return raio;
    }

    public Cor getCorPreenchimento() {
        return corPreenchimento;
    }

    public Vetor2f getPosicao() {
        return posicao;
    }
}

File: ex-14/ex-14_python/circle_shape.py
from cor import Cor
from vetor2f import Vetor2f


class CircleShape:
    def __init__(self, raio):
        self.raio = raio
        self.corPreenchimento = Cor(255, 255, 255)
        self.origem = Vetor2f(0, 0)
        self.posicao = Vetor2f(0, 0)

    def setFillColor(self, cor):
        self.corPreenchimento = cor

    def setOrigin(self, origem):
        self.origem = origem

    def setPosition(self, posicao):
        self.posicao = posicao

    def getRaio(self):
        return self.raio

    def getCorPreenchimento(self):
        return self.corPreenchimento

    def getPosicao(self):
        return self.posicao
